<sdf version='1.7'>
  <world name='default'>

    <scene>
      <ambient>0.4 0.4 0.4 1</ambient>
      <background>0.7 0.7 0.7 1</background>
      <shadows>1</shadows>
    <sky>
    <clouds>
      <speed>12</speed>
    </clouds>
    </sky>
    <fog>
      <type>linear</type>
      <start>10</start>
      <end>500</end>
      <density>1</density>
    </fog>
    </scene>

    <include>
      <uri>model://ground_plane</uri>
    </include>

    <include>
      <uri>model://sun</uri>
    </include>

    <include>
      <uri>model://house_1</uri>
      <pose>-20.0 20.0 0 0 -0 0</pose>
    </include>

    <include>
      <uri>model://house_2</uri>
      <pose>-20.0 -20.0 0 0 -0 0</pose>
    </include>

    <include>
      <uri>model://house_3</uri>
      <pose>20.0 20.0 0 0 -0 0</pose>
    </include>

    <include>
      <uri>model://school</uri>
      <pose>20.0 -20.0 0 0 -0 0</pose>
    </include>

    <model name="camera_0">
      <static>true</static>
      <pose>20 20 12 0 0 -2.2</pose>
      <link name="link">
        <visual name="visual">
          <geometry>
            <box>
              <size>0.3 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <sensor name="camera" type="camera">
          <camera>
            <horizontal_fov>2</horizontal_fov>
            <image>
              <width>2000</width>
              <height>2000</height>
            </image>
            <clip>
              <near>0.1</near>
              <far>100</far>
            </clip>
            <always_on>1</always_on>
            <update_rate>10</update_rate>
            <visualize>1</visualize>
            <save enabled="false">
              <path>../frames/camera0</path>
            </save>
            <lens>
              <type>stereographic</type>
              <scale_to_hfov>1</scale_to_hfov>
              <cutoff_angle>2</cutoff_angle>
            </lens>
          </camera>
          <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
            <alwaysOn>true</alwaysOn>
            <updateRate>10.0</updateRate>
            <cameraName>camera_0</cameraName>
            <imageTopicName>image_raw</imageTopicName>
            <cameraInfoTopicName>camera_0_info</cameraInfoTopicName>
            <frameName>link</frameName>
            <hackBaseline>0.07</hackBaseline>
            <distortionK1>0.0</distortionK1>
            <distortionK2>0.0</distortionK2>
            <distortionK3>0.0</distortionK3>
            <distortionT1>0.0</distortionT1>
            <distortionT2>0.0</distortionT2>
          </plugin>
        </sensor>
        <self_collide>0</self_collide>
        <kinematic>0</kinematic>
      </link>
    </model>

    <model name="camera_1">
      <static>true</static>
      <pose>20 -15 12 0 0 2.2</pose>
      <link name="link">
        <visual name="visual">
          <geometry>
            <box>
              <size>0.3 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <sensor name="camera" type="camera">
          <camera>
            <horizontal_fov>2</horizontal_fov>
            <image>
              <width>2000</width>
              <height>2000</height>
            </image>
            <clip>
              <near>0.1</near>
              <far>100</far>
            </clip>
            <always_on>1</always_on>
            <update_rate>10</update_rate>
            <visualize>1</visualize>
            <save enabled="false">
              <path>../frames/camera1</path>
            </save>
            <lens>
              <type>stereographic</type>
              <scale_to_hfov>1</scale_to_hfov>
              <cutoff_angle>2</cutoff_angle>
            </lens>
          </camera>
          <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
            <alwaysOn>true</alwaysOn>
            <updateRate>10.0</updateRate>
            <cameraName>camera_1</cameraName>
            <imageTopicName>image_raw</imageTopicName>
            <cameraInfoTopicName>camera_1_info</cameraInfoTopicName>
            <frameName>link</frameName>
            <hackBaseline>0.07</hackBaseline>
            <distortionK1>0.0</distortionK1>
            <distortionK2>0.0</distortionK2>
            <distortionK3>0.0</distortionK3>
            <distortionT1>0.0</distortionT1>
            <distortionT2>0.0</distortionT2>
          </plugin>
        </sensor>
        <self_collide>0</self_collide>
        <kinematic>0</kinematic>
      </link>
    </model>

    <model name="camera_2">
      <static>true</static>
      <pose>-20 -20 12 0 0 0.7</pose>
      <link name="link">
        <visual name="visual">
          <geometry>
            <box>
              <size>0.3 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <sensor name="camera" type="camera">
          <camera>
            <horizontal_fov>2</horizontal_fov>
            <image>
              <width>2000</width>
              <height>2000</height>
            </image>
            <clip>
              <near>0.1</near>
              <far>100</far>
            </clip>
            <always_on>1</always_on>
            <update_rate>10</update_rate>
            <visualize>1</visualize>
            <save enabled="false">
              <path>../frames/camera_2</path>
            </save>
            <lens>
              <type>stereographic</type>
              <scale_to_hfov>1</scale_to_hfov>
              <cutoff_angle>2</cutoff_angle>
            </lens>
          </camera>
          <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
            <alwaysOn>true</alwaysOn>
            <updateRate>10.0</updateRate>
            <cameraName>camera_2</cameraName>
            <imageTopicName>image_raw</imageTopicName>
            <cameraInfoTopicName>camera_2_info</cameraInfoTopicName>
            <frameName>link</frameName>
            <hackBaseline>0.07</hackBaseline>
            <distortionK1>0.0</distortionK1>
            <distortionK2>0.0</distortionK2>
            <distortionK3>0.0</distortionK3>
            <distortionT1>0.0</distortionT1>
            <distortionT2>0.0</distortionT2>
          </plugin>
        </sensor>
        <self_collide>0</self_collide>
        <kinematic>0</kinematic>
      </link>
    </model>

    <model name="camera_3">
      <static>true</static>
      <pose>-20 20 12 0 0 -0.7</pose>
      <link name="link">
        <visual name="visual">
          <geometry>
            <box>
              <size>0.3 0.1 0.1</size>
            </box>
          </geometry>
        </visual>
        <sensor name="camera" type="camera">
          <camera>
            <horizontal_fov>2</horizontal_fov>
            <image>
              <width>2000</width>
              <height>2000</height>
            </image>
            <clip>
              <near>0.1</near>
              <far>100</far>
            </clip>
            <always_on>1</always_on>
            <update_rate>10</update_rate>
            <visualize>1</visualize>
            <save enabled="false">
              <path>../frames/camera_3</path>
            </save>
            <lens>
              <type>stereographic</type>
              <scale_to_hfov>1</scale_to_hfov>
              <cutoff_angle>2</cutoff_angle>
            </lens>
          </camera>
          <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
            <alwaysOn>true</alwaysOn>
            <updateRate>10.0</updateRate>
            <cameraName>camera_3</cameraName>
            <imageTopicName>image_raw</imageTopicName>
            <cameraInfoTopicName>camera_3_info</cameraInfoTopicName>
            <frameName>link</frameName>
            <hackBaseline>0.07</hackBaseline>
            <distortionK1>0.0</distortionK1>
            <distortionK2>0.0</distortionK2>
            <distortionK3>0.0</distortionK3>
            <distortionT1>0.0</distortionT1>
            <distortionT2>0.0</distortionT2>
          </plugin>
        </sensor>
        <self_collide>0</self_collide>
        <kinematic>0</kinematic>
      </link>
    </model>

    <model name="drone_0">
      <include>
        <uri>model://iris_with_standoffs</uri>
        <pose>0 0 1 0 0 0</pose>
        <name>drone</name>
      </include>
    </model>

    <model name="drone_1">
      <include>
        <uri>model://iris_with_standoffs</uri>
        <pose>1 0 1 0 0 0</pose>
        <name>drone</name>
      </include>
    </model>

    <gui fullscreen='0'>
      <camera name='user_camera'>
        <pose>-9.6407 49.4187 41.7671 0 0.6058 -1.55978</pose>
        <view_controller>orbit</view_controller>
        <projection_type>perspective</projection_type>
      </camera>
    </gui>

  </world>

</sdf>
<!-- vim: set et fenc=utf-8 ff=unix sts=0 sw=2 ts=2 ft=xml: -->
